Added dark mode

Dhravya opened the pull request:
This PR adds darkMode.

For storing the theme between sessions, I've made use of cookies:
```ts
const theme = cookies().get("theme")
````

I've just use the colors and icons to the best of my knowledge. Please feel free to change them as you like 😄

Here's a demo
![Demo](https://i.dhr.wtf/r/7g476.gif)

steven-tey commented on app/layout.tsx:
```diff
@@ -7,6 +7,7 @@ import { Analytics } from "@vercel/analytics/react";
 import { Metadata } from "next";
 import Toaster from "./toaster";
 import { ReactNode } from "react";
+import { cookies } from "next/dist/client/components/headers";
```
This should be `import { cookies } from 'next/headers'`

Dhravya commented:
@steven-tey I have made the changes :)

Dhravya commented:
As a twitter comment mentioned, it was a bit too dark for the eyes. https://twitter.com/_Eduard26/status/1674449469193039873?s=20 

I tried to soften it
Here, have a look ![https://i.dhr.wtf/r/7o484.gif](https://i.dhr.wtf/r/7o484.gif)

Dhravya commented:
Just realised that checkmarks weren't styled, fixed

![checkmark fix](https://i.dhr.wtf/r/7r487.gif)

Dhravya commented:
The slash and color selector menu were completely black earlier which didn't really look good. I fixed that and 

Also fixed the magic icon not appearing correctly
<img width="307" alt="image" src="https://github.com/steven-tey/novel/assets/63950637/441922a6-cf91-4308-bbd0-28c07f824671">

Dhravya commented:
Hey @steven-tey let me know if there are any more improvements!

steven-tey commented:
I made a switch to using CSS variables instead so we don't need to repeatedly do `dark:` for everything – just `*-stone-*` will do! <picture data-single-emoji=":yayawesome:" title=":yayawesome:"><img class="emoji" width="20" height="auto" src="https://emoji.slack-edge.com/T0CAQ00TU/yayawesome/e9a29a1504448a94.png" alt=":yayawesome:" align="absmiddle"></picture> 

![CleanShot 2023-07-04 at 23 48 04](https://github.com/steven-tey/novel/assets/28986134/cb25be18-7c29-4d34-9f70-cd1582ec9816)
